src/submissions/components/AuthorSubmission.jsx

```diff
@@ -0,0 +1,78 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { Row, Col, Alert } from 'antd';
+import { Formik } from 'formik';
+
+import AuthorForm from '../components/AuthorForm';
+import authorSchema from '../schemas/author';
+import cleanupFormData from '../cleanupFormData';
+import toJS from '../../common/immutableToJS';
+
+const DEFAULT_FORM_DATA = authorSchema.cast();
+
+class AuthorSubmission extends Component {
+  componentDidMount() {
+    this.mounted = true;
+  }
+
+  componentWillUnmount() {
+    this.mounted = false;
+  }
+
+  render() {
+    const { error, initialFormData, onSubmit } = this.props;
+    const initialValues = {
+      ...DEFAULT_FORM_DATA,
+      ...initialFormData,
+    };
+    return (
+      <Row type="flex" justify="center">
+        <Col className="mt3 mb3" span={14}>
+          <div className="mb3 pa3 bg-white">
+            This form allows you to add new author information. All
+            modifications are transferred to{' '}
+            <a
+              target="_blank"
+              rel="noopener noreferrer"
+              href="//inspirehep.net/hepnames"
+            >
+              inspirehep.net/hepnames
+            </a>{' '}
+            upon approval.
+          </div>
+          {error && (
+            <div className="mb3">
+              <Alert message={error.message} type="error" showIcon closable />
+            </div>
+          )}
+          <Formik
+            initialValues={initialValues}
+            validationSchema={authorSchema}
+            onSubmit={async (values, actions) => {
+              const cleanValues = cleanupFormData(values);
+              await onSubmit(cleanValues);
+              if (this.mounted) {
+                actions.setSubmitting(false);
+                window.scrollTo(0, 0);
+              }
+            }}
+            component={AuthorForm}
+          />
+        </Col>
+      </Row>
+    );
+  }
+}
+
+AuthorSubmission.propTypes = {
+  error: PropTypes.objectOf(PropTypes.any), // must have 'message'
```

Comment:
can we use ``.shape``?

Reply:
I think `shape` is strict and it does not ignore unknown properties so if you pass more things, it will complain.

Reply:
but it's ugly to have an instruction like that

Reply:
can we define the shape of the error?

Reply:
once we have stable api errors we can define